bokehjs/src/coffee/models/canvas/canvas.coffee

```diff
@@ -47,15 +47,23 @@ class CanvasView extends BokehView
     ctx = canvas_el[0].getContext('2d')
     return ctx
 
+  need_render: () ->
+    # Public method so the plot can determine if size/dpi changed
+    width = @mget('width')
+    height = @mget('height')
```

Comment:
Yes: change your browser zoom from 100% to say 150%. The text (and glyphs) should remain crisp. If it only switches after manipulating the plot, it did not work.

Testing this with Selenium is probably a bit difficult? Maybe we could test that the `pixel_ratio` is indeed changed or something. Selenium does seem [capable of zooming](http://stackoverflow.com/questions/15024756).